contracts/CEtherDelegateTempExploitAccounting.sol

```diff
@@ -0,0 +1,162 @@
+pragma solidity ^0.5.17;
+
+import "./CEther.sol";
+
+/**
+ * @title Compound's CEtherDelegate Contract
+ * @notice CTokens which wrap Ether and are delegated to
+ * @author Compound
+ */
+contract CEtherDelegateTempExploitAccounting is CDelegateInterface, CEther {
+    /**
+     * @notice Construct an empty delegate
+     */
+    constructor() public {}
+
+    /**
+     * @notice Called by the delegator on a delegate to initialize it for duty
+     * @param data The encoded bytes data for any initialization
+     */
+    function _becomeImplementation(bytes calldata data) external {
+        // Shh -- currently unused
+        data;
+
+        // Shh -- we don't ever want this hook to be marked pure
+        if (false) {
+            implementation = address(0);
+        }
+
+        require(msg.sender == address(this) || hasAdminRights(), "!self");
+        require(accrueInterest() == uint(Error.NO_ERROR), "accrue interest failed");
+
+        // Get secondary accounts from data
+        (address[] memory secondaryAccounts) = abi.decode(data, (address[]));
+        uint256 secondaryAccountsBorrowBalance = 0;
+
+        for (uint256 i = 0; i < secondaryAccounts.length; i++) {
+            address secondaryAccount = secondaryAccounts[i];
+
+            // Get account #2 borrow balance
+            uint256 secondaryAccountBorrowBalance = div_(mul_(accountBorrows[secondaryAccount].principal, borrowIndex), accountBorrows[secondaryAccount].interestIndex);
+            secondaryAccountsBorrowBalance = add_(secondaryAccountsBorrowBalance, secondaryAccountBorrowBalance);
+
+            // Set account #2 borrow balance to 0
+            accountBorrows[secondaryAccount].principal = 0;
```

Comment:
why is the borrower index not set here? like in https://github.com/Rari-Capital/compound-protocol/blob/fuse-final/contracts/CToken.sol#L934

Reply:
because principal is 0

Reply:
what happens if the hacker deposits again - will the borrower index get reset to the correct amount?

Reply:
I'd prefer to simulate more closely the actual steps outlined in herehttps://hackmd.io/LXQlYm-7Se26BMPcDKTYjw

Reply:
fixed